### PR TITLE
wafHook: don't add cross compilation flags

### PR DESCRIPTION
**Pull request by alyssais**
## Description of changes

**This should wait until breaking changes are allowed again.**

These flags are not part of waf, they're custom flags that are not widely implemented.  More packages are broken because of these flags being added than actually recognise them.

Of the packages in Nixpkgs that directly depend on wafHook that we can attempt to cross compile (i.e. all their dependencies cross compile), 5 already successfully cross compile and recognise these flags, 2 already successfully cross compile because they have been opted out of these flags, 2 don't cross compile successfully for reasons unrelated to these flags, and for the remaining 8, the only thing stopping them cross compiling successfully is that they are being passed these flags that they don't recognise.

All of the five successfully cross-compiling packages that do recognise these flags are samba projects: ldb, talloc, tdb, tevent, and samba4.  So this isn't a general waf convention, just a samba one. It therefore doesn't make sense to set these flags by default.  They should just be included in the expressions for each samba project, like all the other quirks common to samba build systems.

This change fixes cross compilation of the following packages:

  blockhash fomp ganv ndn-cxx mda_lv2 pflask raul saldl

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by symphorien** [comment]
> This change fixes cross compilation of the following packages:

> blockhash fomp ganv ndn-cxx mda_lv2 pflask raul saldl

fomp-aarch64-unknown-linux-gnu-1.2.2 fails with a cross related error:
```
/nix/store/gy66lj8qszy07whk7r1cgmihyrppx8gv-waf-setup-hook/nix-support/setup-hook: line 26: python: command not found
```

apart from that lgtm, thanks for the in-depth investigation.

**Comment by alyssais** [comment]
Fixed in the commit message; thanks.

**Review by symphorien — APPROVED** [review]

File: doc/hooks/waf.section.md [hunk 1]
```diff
@@ -20,10 +20,6 @@ If `wafPath` doesn't exist, then `wafHook` will copy the `waf` provided from Nix
 
 Controls the flags passed to waf tool during build and install phases. For settings specific to build or install phases, use `wafBuildFlags` or `wafInstallFlags` respectively.
 
-#### `dontAddWafCrossFlags` {#dont-add-waf-cross-flags}
```
Comment (AndersonTorres) [inline doc/hooks/waf.section.md:23]: Hum, since it changes the documentation, should we include a note on release logs?

Comment (alyssais) [reply doc/hooks/waf.section.md:23]: Could do, but we don't have release notes for 24.11 yet.  I think that gets created as part of the release management process.

**Pull request merged**